Unable to do storage migration on Androids 5.1-7.1

#### Software and hardware versions 
Collect v1.28 master.x, Android v5.1-7.1, device used...

#### Problem description
I have encountered a weird issue on Androids 5.1 and 7.1 @mmarciniak90 confirmed also that it's visible on Android 7.0. Storage migration fails:
![ezgif com-video-to-gif (39)](https://user-images.githubusercontent.com/33342488/95326139-652b6900-08a2-11eb-8c24-795227a81f68.gif)

stacktrace:
`2020-10-07 12:55:26.698 14590-14590/org.odk.collect.android E/SQLiteDatabase: Failed to open database '/storage/emulated/0/odk/metadata/instances.db'.
    android.database.sqlite.SQLiteCantOpenDatabaseException: unknown error (code 14): Could not open database
        at android.database.sqlite.SQLiteConnection.nativeOpen(Native Method)
        at android.database.sqlite.SQLiteConnection.open(SQLiteConnection.java:209)
        at android.database.sqlite.SQLiteConnection.open(SQLiteConnection.java:193)
        at android.database.sqlite.SQLiteConnectionPool.openConnectionLocked(SQLiteConnectionPool.java:463)
        at android.database.sqlite.SQLiteConnectionPool.open(SQLiteConnectionPool.java:185)
        at android.database.sqlite.SQLiteConnectionPool.open(SQLiteConnectionPool.java:177)
        at android.database.sqlite.SQLiteDatabase.openInner(SQLiteDatabase.java:808)
        at android.database.sqlite.SQLiteDatabase.open(SQLiteDatabase.java:793)
        at android.database.sqlite.SQLiteDatabase.openDatabase(SQLiteDatabase.java:696)
        at android.database.sqlite.SQLiteDatabase.openDatabase(SQLiteDatabase.java:671)
        at org.odk.collect.android.database.InstancesDatabaseHelper.databaseNeedsUpgrade(InstancesDatabaseHelper.java:226)
        at org.odk.collect.android.provider.InstanceProvider.getDbHelper(InstanceProvider.java:69)
        at org.odk.collect.android.provider.InstanceProvider.query(InstanceProvider.java:122)
        at android.content.ContentProvider.query(ContentProvider.java:1020)
        at android.content.ContentProvider$Transport.query(ContentProvider.java:239)
        at android.content.ContentResolver.query(ContentResolver.java:532)
        at android.content.ContentResolver.query(ContentResolver.java:474)
        at org.odk.collect.android.dao.InstancesDao.getInstancesCursor(InstancesDao.java:241)
        at org.odk.collect.android.dao.InstancesDao.getFinalizedInstancesCursor(InstancesDao.java:138)
        at org.odk.collect.android.activities.MainMenuActivity.countSavedForms(MainMenuActivity.java:369)
        at org.odk.collect.android.activities.MainMenuActivity.onResume(MainMenuActivity.java:276)
        at android.app.Instrumentation.callActivityOnResume(Instrumentation.java:1291)
        at android.app.Activity.performResume(Activity.java:6791)
        at android.app.ActivityThread.performResumeActivity(ActivityThread.java:3427)
        at android.app.ActivityThread.handleResumeActivity(ActivityThread.java:3490)
        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2749)
        at android.app.ActivityThread.-wrap12(ActivityThread.java)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1490)
        at android.os.Handler.dispatchMessage(Handler.java:102)
        at android.os.Looper.loop(Looper.java:154)
        at android.app.ActivityThread.main(ActivityThread.java:6165)
        at java.lang.reflect.Method.invoke(Native Method)`

#### Steps to reproduce the problem
I was able to reproduce just by installing Collect and try storage migration. Might be hard to find more steps.

#### Expected behavior
Storage migration should be successful. 

#### Other information 
odk folder: [odkfolder.zip](https://github.com/getodk/collect/files/5340358/odkfolder.zip)
folder created after failed migration: [postmigrationtry.zip](https://github.com/getodk/collect/files/5340363/postmigrationtry.zip)


@getodk-bot label "Specific to Android version"

I was able to reproduce the issue. This would be a better stacktrace:
```
2020-10-07 13:58:16.019 4428-4523/org.odk.collect.android W/StorageMigrator: java.lang.NoClassDefFoundError: Failed resolution of: [Ljava/nio/file/CopyOption;
        at org.apache.commons.io.FileUtils.copyDirectory(FileUtils.java:659)
        at org.apache.commons.io.FileUtils.copyDirectory(FileUtils.java:555)
        at org.apache.commons.io.FileUtils.copyDirectory(FileUtils.java:521)
        at org.odk.collect.android.storage.migration.StorageMigrator.moveAppDataToScopedStorage(StorageMigrator.java:99)
        at org.odk.collect.android.storage.migration.StorageMigrator.migrate(StorageMigrator.java:74)
        at org.odk.collect.android.storage.migration.StorageMigrator.performStorageMigration(StorageMigrator.java:60)
        at org.odk.collect.android.storage.migration.StorageMigrationService.lambda$null$0$StorageMigrationService(StorageMigrationService.java:42)
        at org.odk.collect.android.storage.migration.-$$Lambda$StorageMigrationService$eJ0q_ANeq_5ItRtUVwPrCUvD2Dc.apply(lambda)
        at org.odk.collect.android.backgroundwork.ReentrantLockChangeLock.withLock(ReentrantLockChangeLock.java:13)
        at org.odk.collect.android.storage.migration.StorageMigrationService.lambda$onHandleIntent$1$StorageMigrationService(StorageMigrationService.java:40)
        at org.odk.collect.android.storage.migration.-$$Lambda$StorageMigrationService$XMgfn0INwsgVjjTmLXhqsvBC5Zo.apply(lambda)
        at org.odk.collect.android.backgroundwork.ReentrantLockChangeLock.withLock(ReentrantLockChangeLock.java:13)
        at org.odk.collect.android.storage.migration.StorageMigrationService.onHandleIntent(StorageMigrationService.java:39)
        at android.app.IntentService$ServiceHandler.handleMessage(IntentService.java:68)
        at android.os.Handler.dispatchMessage(Handler.java:102)
        at android.os.Looper.loop(Looper.java:154)
        at android.os.HandlerThread.run(HandlerThread.java:61)
     Caused by: java.lang.ClassNotFoundException: Didn't find class "java.nio.file.CopyOption" on path: DexPathList[[zip file "/system/framework/com.google.android.maps.jar", zip file "/data/app/org.odk.collect.android-2/base.apk"],nativeLibraryDirectories=[/data/app/org.odk.collect.android-2/lib/x86, /data/app/org.odk.collect.android-2/base.apk!/lib/x86, /system/lib, /vendor/lib]]
```

--- a/collect_app/build.gradle
+++ b/collect_app/build.gradle
@@ -231,7 +231,10 @@ dependencies {
     implementation 'com.github.chanmratekoko:myanmar-calendar:1.0.6.RC3'
     implementation 'bikramsambat:bikram-sambat:1.1.0'
     implementation "com.rarepebble:colorpicker:3.0.1"
-    implementation "commons-io:commons-io:2.8.0"
+
+    // Updating would require minSdkVersion >=26
+    implementation "commons-io:commons-io:2.6"
+
     implementation "net.sf.opencsv:opencsv:2.4"
     implementation("org.getodk:javarosa:3.0.2") {
         exclude group: 'joda-time'

